plugins/events/create.js

```diff
@@ -88,54 +117,75 @@ module.exports = () => ({
                         throw boom.unauthorized('Token does not have permission to this pipeline');
                     }
 
+                    let token;
+                    let scmConfig;
+                    let prInfo;
+
                     // Check if user has push access
                     // eslint-disable-next-line consistent-return
                     return user.getPermissions(pipeline.scmUri)
                         .then((permissions) => {
-                            if (!permissions.push) {
-                                const newAdmins = pipeline.admins;
-
-                                delete newAdmins[username];
-                                // This is needed to make admins dirty and update db
-                                pipeline.admins = newAdmins;
-
-                                return pipeline.update()
-                                    .then(() => {
-                                        throw boom.unauthorized(`User ${username} `
-                                        + 'does not have push permission for this repo');
-                                    });
+                            if (!permissions.push && !prNum) {
+                                return updateAdmins({ pipeline, isAdmin: false, username });
```

Comment:
feel the `updateAdmins` can just take in the `permissions` object. And for the non-pr case, it can be all done here in this block. then you don't need the part in the end: https://github.com/screwdriver-cd/screwdriver/blob/b81432caaae7b4f467ccf3cb0c4086cc6d29ba08/plugins/events/create.js#L180

```js
if (!prNum) {
     return updateAdmins({ pipeline, permissions, username });
}
```

Reply:
same for the pr case